enhancements/accelerators/out-of-tree-drivers.md

```diff
@@ -0,0 +1,418 @@
+ 
+#### Story 2: Multiple DriverContainers
+As an administrator I want to enable a hardware stack to enable a specific functionality with several kernel modules. These kernel modules may have an order and the procedure enabling them needs to expose a way to model this dependency.  It may even be the case that a specific module needs kernel modules loaded that are already installed on the node.  The intended clusters are either behind a proxy or completely disconnected and hence the anticipated procedure has to work in these environments too. 
+
+#### Story 3: Day-2 DriverContainer Accelerator (short-term)
+As a vendor of a hardware accelerator I want a procedure to enable the accelerator on OpenShift no matter which underlying OS is running on the nodes. The life-cycle of the drivers should be fully managed with the ability to upgrade and downgrade drivers for the accelerator. It should support all kernel versions (major, minor, z-stream) and handle driver errors gracefully. Uninstalling the drivers should not leave any trace of the previous installation (keep the node as clean as possible). The driver will not be upstreamed to the mainline kernel, which means it will always be out-of-tree. 
+
+#### Story 4: Day-1 DriverContainer (long-term)
+#### Story 5: Multiple Driver Containers with Artifacts  (long-term)
+As an administrator I want to enable a specific vendor stack. I need to build kernel modules that are dependent on each other during the build and at the time of loading. Specific build artifacts need to be available during the build not for loading. These artifacts can be available  in another DriverContainer or extracted during runtime. 
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+
+[BuildConfigs do not support volume mounts](https://issues.redhat.com/browse/DEVEXP-17), cannot be used for artifacts on a hostPath. If we have all artifacts stored in a container image, BuildConfig can/could be leveraged for building. Where no build artifacts are needed a BuildConfig is a first choice, because of the functionality it provides (triggers, source, output, etc).
+One of the most important things, we have no issues with SELinux, because we are interacting with the same SELinux contexts (container_file_t). Accessing host devices, libraries, or binaries from a container breaks the containment
+as we have to allow containers to access host labels. 
+ 
+For DriverContainers to build the kernel modules we need entitlements first. The e2e story is described here: [https://bit.ly/2XjZq5D](https://bit.ly/2XjZq5D)
```

Comment:
Generally speaking I think this is a good idea. Though, as noted above, OCP should have the needed kernel deps available to build kmods as they will be available in the machine-os-content ... but it's also fair to denote entitlement as a need especially when speaking to RHEL outside of OCP.

Reply:
Some driver-containers need more RPMs to be installed then just kernel-devel and kernel-headers. Some containers have daemons or other auxiliary tools that have dependencies on e.g. elf-utils.